control-plane-operator/controllers/hostedcontrolplane/cloud/azure/providerconfig.go

```diff
@@ -43,8 +43,19 @@ func ReconcileCloudConfigWithCredentials(secret *corev1.Secret, hcp *hyperv1.Hos
 		return err
 	}
 
-	cfg.AADClientID = hcp.Spec.Platform.Azure.ManagedIdentities.ControlPlane.CloudProvider.ClientID
-	cfg.AADClientCertPath = config.ManagedAzureCertificatePath + hcp.Spec.Platform.Azure.ManagedIdentities.ControlPlane.CloudProvider.CertificateName
+	// Get the ClientID and CertificateName config for csi drivers in ARO
```

Comment:
can we have a unit test for this func?

Reply:
Can we please make sure this change is also applied to cpo v2/ cc @muraee @bryan-cox

Reply:
https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/v2/cloud_controller_manager/azure/config.go#L60

Reply:
TBH, I could not understand the cpo v2 design how to handle the cloud secrets, I saw it introduces the `azure-credential-information` secret while seems no logic handle the csi drivers secrets. Not sure how to apply the fix also to cpo v2. @bryan-cox Could you please help give some info? Thank you!

Reply:
true, this will be part for the storage v2 which is not yet refactored. You don't need to do anything extra for now in this PR.

Reply:
@muraee Thanks for the checking and clarifying!